### PR TITLE
Improved viewer screenshot handling

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Minor change to add a screenshot directory and incremental screenshot saving for viewer.

## How Has This Been Tested

Local testing.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/utils/viewer/viewer.cpp [hunk 1]
```diff
@@ -783,8 +784,12 @@ void Viewer::keyPressEvent(KeyEvent& event) {
...
       break;
     case KeyEvent::Key::I:
-      Mn::DebugTools::screenshot(Mn::GL::defaultFramebuffer,
-                                 "test_image_save.png");
+      if (!Cr::Utility::Directory::exists("screenshots/")) {
+        Cr::Utility::Directory::mkpath("screenshots/");
+      }
+      Mn::DebugTools::screenshot(
+          Mn::GL::defaultFramebuffer,
+          "screenshots/" + std::to_string(savedFrames++) + ".png");
```
Comment (eundersander) [inline src/utils/viewer/viewer.cpp:792]: Nit: it'd be nice if this feature didn't overwrite existing screenshots from previous runs. Can you think of any easy way to improvise unique filenames? My Linux screenshot tool uses timestamps.

**Review by eundersander — APPROVED** [review]
I made a small suggestion, but anyway nice feature! LGTM!

File: src/utils/viewer/viewer.cpp (hunk 1 above)
Comment (bigbike) [reply src/utils/viewer/viewer.cpp:792]: @eundersander: An int64 number as the suffix? The largest number means the latest screenshot? :)

**Review by Skylion007 — APPROVED** [review]

File: src/utils/viewer/viewer.cpp (hunk 1 above)
Comment (Skylion007) [reply src/utils/viewer/viewer.cpp:792]: int64??? We expect people to take 1 billion+ images in a single folder. Most setups wouldn't have that many inodes across the whole filesystem.

Comment (aclegg3) [reply src/utils/viewer/viewer.cpp]: I'm now caching the date-time when viewer is started. This is used to create a directory for screenshots to keep them separated on each run.

**Review by jturner65 — APPROVED** [review]
LGTM

**Pull request merged**